presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -197,7 +199,9 @@ else if (type instanceof ObjectType) {
 
         ConnectorTableMetadata tableMetadata = getTableMetadata(session, tableHandle);
         for (ColumnMetadata column : tableMetadata.getColumns()) {
-            results.put(column.getName(), new ElasticsearchColumnHandle(column.getName(), column.getType()));
+            results.put(column.getName(), new ElasticsearchColumnHandle(
+                    column.getProperties().getOrDefault(ORIGINAL_NAME, column.getName()).toString(),
```

Comment:
There is no reason to accept anything other than `String` here.

```suggestion
                    (String) column.getProperties().getOrDefault(ORIGINAL_NAME, column.getName()),
```